hedera-node/src/main/java/com/hedera/services/txns/network/FreezeTransitionLogic.java

```diff
@@ -86,9 +87,19 @@ public void doStateTransition() {
 
 	public ResponseCodeEnum validate(TransactionBody freezeTxn) {
 		var op = freezeTxn.getFreeze();
-		if (!isValidTime(op.getStartHour(), op.getStartMin()) || !isValidTime(op.getEndHour(), op.getEndMin())) {
-			return INVALID_FREEZE_TRANSACTION_BODY;
+
+		if(op.hasStartTime()) {
+			Timestamp txnStartTime = freezeTxn.getTransactionID().getTransactionValidStart();
+			if(!isValidTimestamp(op.getStartTime(), txnStartTime)) {
+				return INVALID_FREEZE_TRANSACTION_BODY;
+			}
 		}
+		else {
+			if (!isValidTime(op.getStartHour(), op.getStartMin()) || !isValidTime(op.getEndHour(), op.getEndMin())) {
+				return INVALID_FREEZE_TRANSACTION_BODY;
```

Comment:
can we move the checks in line 98 and 91,93 into a new method so that we will have only one `return INVALID_FREEZE_TRANSACTION_BODY;` here instead of two ?

Like
``` 
if(!isValidFreeze(op, txnStartTime)) {
    return INVALID_FREEZE_TRANSACTION_BODY;
}

...

private boolean isValidFreeze(op, txnStartTime) {
    return ( op..hasStartTime() && !isValidTimestamp(op.getStartTime(), txnStartTime) ) || 
                  !isValidTime(op.getStartHour(), op.getStartMin()) || !isValidTime(op.getEndHour(), op.getEndMin())
}
```